commit 4ffecdbf7d68e7d9e121e3acf458780d48a661a4
Merge: 2dfdf4b 067b509
Author: Bo Herrmannsen <bo.herrmannsen@gmail.com>
Date:   Tue Jan 6 13:47:49 2015 +0100

    Merge pull request #1313 from wgm4321/Development
    
    Re-enable filament menu with recalc fix when enabling/disabling mm3

diff --git a/Marlin/ultralcd.cpp b/Marlin/ultralcd.cpp
index 42d4979..d790266 100644
--- a/Marlin/ultralcd.cpp
+++ b/Marlin/ultralcd.cpp
@@ -880,7 +880,7 @@ static void lcd_control_volumetric_menu()
 	START_MENU();
 	MENU_ITEM(back, MSG_CONTROL, lcd_control_menu);
 
-	MENU_ITEM_EDIT(bool, MSG_VOLUMETRIC_ENABLED, &volumetric_enabled);
+	MENU_ITEM_EDIT_CALLBACK(bool, MSG_VOLUMETRIC_ENABLED, &volumetric_enabled, calculate_volumetric_multipliers);
 
 	if (volumetric_enabled) {
 		MENU_ITEM_EDIT_CALLBACK(float43, MSG_FILAMENT_SIZE_EXTRUDER_0, &filament_size[0], DEFAULT_NOMINAL_FILAMENT_DIA - .5, DEFAULT_NOMINAL_FILAMENT_DIA + .5, calculate_volumetric_multipliers);
@@ -1114,6 +1114,11 @@ static void menu_action_setting_edit_bool(const char* pstr, bool* ptr)
 {
     *ptr = !(*ptr);
 }
+static void menu_action_setting_edit_callback_bool(const char* pstr, bool* ptr, menuFunc_t callback)
+{
+	menu_action_setting_edit_bool(pstr, ptr);
+	(*callback)();
+}
 #endif//ULTIPANEL
 
 /** LCD API **/
